lib/tile_bag.rb

```diff
@@ -0,0 +1,35 @@
+require 'pry'
+
+module Scrabble
+  class TileBag
+
+    attr_reader :tiles, :number_of_tiles
+
+    LETTER_QUANTITIES = {
```

Comment:
This is great for the letter quantities, but you're using this hash to store the tiles in the bag.  This is problematic for a couple of reasons.
1.  All instances of `TileBag` are **sharing** this constant.  So if you remove a "z" from one `TileBag` instance, you've removed it from them all.
1.  This structure makes any letter  **equally** likely to be drawn from the bag.  However there is only 1 "z" and 12 "E"'s.